### PR TITLE
add node:dns resolver class stub implementation

### PR DESCRIPTION
**Pull request by anonrig**
Adds `node:dns` Resolver class stub implementation

### CODE REVIEW

File: src/workerd/api/node/tests/dns-nodejs-test.js [hunk 1]
```diff
@@ -28,6 +28,9 @@ export const functionsExist = {
     for (const fn of syncFns) {
       strictEqual(typeof dns[fn], 'function');
     }
+
+    ok(dns.Resolver);
```
Comment (mikea) [inline src/workerd/api/node/tests/dns-nodejs-test.js:32]: do you want to call a method (and get exception) just for completeness sake?

File: src/workerd/api/node/dns.h [hunk 2]
```diff
@@ -117,11 +160,15 @@ class DnsUtil final: public jsg::Object {
...
+
+    // Classes
+    JSG_NESTED_TYPE_NAMED(DnsResolver, Resolver);
   }
 };
 
 #define EW_NODE_DNS_ISOLATE_TYPES                                                                  \
-  api::node::DnsUtil, api::node::DnsUtil::LookupOptions, api::node::DnsUtil::Resolve4Options,      \
-      api::node::DnsUtil::Resolve6Options
+  api::node::DnsUtil, api::node::DnsUtil::LookupOptions,                                           \
```
Comment (mikea) [inline src/workerd/api/node/dns.h:170]: will this remove `Resolve4Options` from api? Is it a breaking change? it doesn't seem so, just trying to be careful here.

File: src/workerd/api/node/dns.h [hunk 3]
```diff
@@ -88,10 +63,78 @@ class DnsUtil final: public jsg::Object {
 
   using ReverseCallback = jsg::Function<void(void)>;
   void reverse(jsg::Lock& js, kj::String ip, ReverseCallback callback);
+};
+
+class DnsResolver final: public DnsResolverBase, public jsg::Object {
```
Comment (mikea) [inline src/workerd/api/node/dns.h:68]: I know you node people know these by heart but I would _love_ to see comments like:

// https://nodejs.org/api/dns.html#class-dnsresolver

on these classes.

**Review by mikea — APPROVED** [review]

File: src/workerd/api/node/dns.h (hunk 2 above)
Comment (anonrig) [reply src/workerd/api/node/dns.h:170]: I just renamed it to ResolveOptions. No breaking changes. This API is not implemented yet. All functions have a stub implementation.

File: src/workerd/api/node/dns.h (hunk 3 above)
Comment (anonrig) [reply src/workerd/api/node/dns.h:68]: Definitely.

**Review by jasnell — APPROVED** [review]

**Pull request closed without merging**